iot-hub/Samples/service/ServiceClientSample/ServiceClientSample.cs

```diff
@@ -50,13 +54,50 @@ public async Task RunSampleAsync(TimeSpan runningTime)
 
         }
 
+        private async Task ReceiveMessageFeedbacksAsync(CancellationToken token)
+        {
+            _logger.LogInformation("Starting to listen to feedback messages");
+
+            while (!token.IsCancellationRequested)
+            {
+                var feedbackReceiver = _serviceClient.GetFeedbackReceiver();
+
+                try
+                {
+                    FeedbackBatch feedbackMessages = await feedbackReceiver.ReceiveAsync();
+                    if (feedbackMessages != null)
+                    {
+                        _logger.LogInformation("New Feedback received:");
+                        _logger.LogInformation($"\tEnqueue Time: {feedbackMessages.EnqueuedTime}");
+                        _logger.LogInformation($"\tNumber of messages in the batch: {feedbackMessages.Records.Count()}");
+                        foreach (FeedbackRecord feedbackRecord in feedbackMessages.Records)
+                        {
+                            _logger.LogInformation($"\tDevice {feedbackRecord.DeviceId} acted on message: {feedbackRecord.OriginalMessageId} with status: {feedbackRecord.StatusCode}");
+                        }
+
+                        await feedbackReceiver.CompleteAsync(feedbackMessages);
+                    }
+                }
+                catch (Exception e)
+                {
+                    _logger.LogError($"Unexpected error, will need to reinitialize the client: {e}");

```

Comment:
nit: we wouldn't need to reinitialize for all exceptions, right? let's add a transient check in here.

Reply:
```suggestion
                    _logger.LogError($"Unexpected error - will need to reinitialize the client: {e}");

```